tests/commands/test_build.py

```diff
@@ -360,10 +361,13 @@ def test_build_basic_complete_structure(tmp_path):
     build_dir = tmp_path / BUILD_DIRNAME
     build_dir.mkdir()
 
-    # the metadata
-    metadata = tmp_path / 'metadata.yaml'
-    with metadata.open('wb') as fh:
-        fh.write(b'lot of yaml config')
+    # the metadata (save it and restore to later check)
+    metadata_data = {'name': 'name-from-metadata'}
+    metadata_file = tmp_path / 'metadata.yaml'
+    with metadata_file.open('wt', encoding='ascii') as fh:
+        yaml.dump(metadata_data, fh)
```

Comment:
It seems a little odd to write it out and then read it again. Why not:
```
with metadata_file.open(...) as fh:
  raw = yaml.dump(metadata_data)
  fh.write(raw)
```

Reply:
Ah, didn't know about that. I always wondering why `yaml` didn't have a `dumps` function.

Thanks! Improved.